src/snowflake/snowpark/open_telemetry_prototype.py

```diff
@@ -0,0 +1,76 @@
+import inspect
+from typing import Dict, Union
+
+from opentelemetry.sdk.resources import SERVICE_NAME, Resource
+
+from opentelemetry import trace
+from opentelemetry.sdk.trace import TracerProvider
+from opentelemetry.sdk.trace.export import BatchSpanProcessor, ConsoleSpanExporter
+
+from opentelemetry import metrics
+from opentelemetry.sdk.metrics import MeterProvider
+from opentelemetry.sdk.metrics.export import PeriodicExportingMetricReader, ConsoleMetricExporter
+
+
+resource = Resource(attributes={
+    SERVICE_NAME: "snowpark-python-open-telemetry"
+})
+
+traceProvider = TracerProvider(resource=resource)
+processor = BatchSpanProcessor(ConsoleSpanExporter())
+traceProvider.add_span_processor(processor)
+
+trace.set_tracer_provider(traceProvider)
+
+tracer = trace.get_tracer("df.filter.tracer")
+
+reader = PeriodicExportingMetricReader(ConsoleMetricExporter())
+meterProvider = MeterProvider(resource=resource, metric_readers=[reader])
+metrics.set_meter_provider(meterProvider)
+
+
+def open_telemetry(name):
+    def open_telemetry_decorator(func):
+        def wrapper(*df, **params):
+            # get complete parameter list of the function
+            dataframe, parameters = parameter_decoder(df, params, func)
+            with tracer.start_as_current_span(name) as cur_span:
+                # store parameters in span
+                cur_span.set_attribute("function_parameters", str(parameters))
+
+                # store execution location in span
+                frame_info = inspect.stack()[-1]
+                cur_span.set_attribute("execute_location", f"{frame_info.filename}#{frame_info.lineno}")
```

Comment:
Opentelemetry has standard attributes for this, see https://opentelemetry.io/docs/specs/semconv/attributes-registry/code/